src/commands/createFunction/scriptSteps/ScriptFunctionCreateStep.ts

```diff
@@ -56,8 +57,26 @@ export class ScriptFunctionCreateStep extends FunctionCreateStepBase<IScriptFunc
 
         const language: ProjectLanguage = nonNullProp(context, 'language');
         const fileName: string | undefined = getScriptFileNameFromLanguage(language);
+
+        await this.installDependencies(language, context.projectPath);
         return fileName ? path.join(functionPath, fileName) : functionJsonPath;
     }
 
+    private async installDependencies(language: string, projectPath: string): Promise<void> {
+        switch (language) {
+            case ProjectLanguage.CSharpScript:
+            case ProjectLanguage.FSharpScript:
+            case ProjectLanguage.JavaScript:
```

Comment:
This `switch` statement isn't implemented correctly; this will essentially do `nodeUtils.installDependencies(projectPath)` for *all* project types.
This is what you're looking for:
```typescript
        switch (language) {
            case ProjectLanguage.JavaScript:
            case ProjectLanguage.TypeScript:
                await nodeUtils.installDependencies(projectPath);
                break;
            case ProjectLanguage.CSharpScript:
            case ProjectLanguage.FSharpScript:
            case ProjectLanguage.PowerShell:
            case ProjectLanguage.Python:
            default:
                break;
        }
```

Reply:
Oh right, that makes sense, thank you for catching that!

Reply:
I think I'd rather there be a separate `JavaScriptFunctionCreateStep` that handles this instead of having a switch case for generic `ScriptFunctionCreateStep`.  We already have a `TypeScruptFunctionCreateStep` as it is.